Nightly CI seems to have been failing for a while in random jobs

Nightly CI seems to be failing where random jobs show the error below. From what I can see the first occurred on 16th Aug which would be following the release of Qiskit 1.2.0 the day before.
```
      File "D:\a\qiskit-machine-learning\qiskit-machine-learning\test\circuit\library\test_raw_feature_vector.py", line 63, in test_fully_bound
    self.assertEqual(bound.decompose(), ref)

      File "c:\hostedtoolcache\windows\python\3.8.10\x64\lib\unittest\case.py", line 912, in assertEqual
    assertion_func(first, second, msg=msg)

      File "c:\hostedtoolcache\windows\python\3.8.10\x64\lib\unittest\case.py", line 905, in _baseAssertEqual
    raise self.failureException(msg)

    AssertionError: <qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x0000029754FC25E0> != <qiskit.circuit.quantumcircuit.QuantumCircuit object at 0x000002975636DBE0>
```
1.2.0 came with changes to have more logic in Rust for performance e.g. circuit data and there were changes to Initialize and StateVector as well which I think Raw Feature ends up using. But the failure seems random as in it passes in most jobs.

--- a/qiskit_machine_learning/circuit/library/raw_feature_vector.py
+++ b/qiskit_machine_learning/circuit/library/raw_feature_vector.py
@@ -178,7 +178,7 @@ class ParameterizedInitialize(Instruction):
         # normalize
         norm = np.linalg.norm(cleaned_params)
         normalized = cleaned_params if np.isclose(norm, 1) else cleaned_params / norm
-Fix RawFeatire
+
         circuit = QuantumCircuit(self.num_qubits)
         circuit.initialize(normalized, range(self.num_qubits))
         self.definition = circuit

--- a/qiskit_machine_learning/circuit/library/raw_feature_vector.py
+++ b/qiskit_machine_learning/circuit/library/raw_feature_vector.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2020, 2023.
+# (C) Copyright IBM 2020, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -176,8 +176,9 @@ class ParameterizedInitialize(Instruction):
                 raise QiskitError("Cannot define a ParameterizedInitialize with unbound parameters")
 
         # normalize
-        normalized = np.array(cleaned_params) / np.linalg.norm(cleaned_params)
-
+        norm = np.linalg.norm(cleaned_params)
+        normalized = cleaned_params if np.isclose(norm, 1) else cleaned_params / norm
+Fix RawFeatire
         circuit = QuantumCircuit(self.num_qubits)
         circuit.initialize(normalized, range(self.num_qubits))
         self.definition = circuit

